core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -205,20 +226,20 @@ class BrokerToControllerRequestThread(
 
   private[server] def handleResponse(request: BrokerToControllerQueueItem)(response: ClientResponse): Unit = {
     if (response.wasDisconnected()) {
-      activeController = None
+      updateControllerAddress(None)
       requestQueue.putFirst(request)
     } else if (response.responseBody().errorCounts().containsKey(Errors.NOT_CONTROLLER)) {
       // just close the controller connection and wait for metadata cache update in doWork
-      networkClient.disconnect(activeController.get.idString)
-      activeController = None
+      networkClient.disconnect(activeControllerAddress().get.idString)
```

Comment:
This is another slippery looking case. Can we just rewrite this as a `foreach` so that we don't need to worry about it?